elementary/gala
# Implement ScreenSaver interfaces

Fixes #716 

This implements a screen saver/screen locker embedded in the compositor, effectively replacing `light-locker`.

To test:
* Install this branch
* Disable `light-locker` by renaming it's autostart file at `/etc/xdg/autostart/light-locker.desktop` to `/etc/xdg/autostart/light-locker.desktop.disabled`
* Reboot
* Set `/org/pantheon/desktop/gala/behavior/lock-on-suspend` and `/org/pantheon/desktop/gala/behavior/lock-on-screen-off` to your desired values. 

Notes:
The "turn off display when inactive for" option in the power plug still controls the session timeout (i.e. when the screensaver activates), and the suspend settings should work as expected.

The two locking options in the privacy plug will no longer do anything due to them controlling `light-locker` dconf keys, hence the manual setting of the two keys above. We'll need to follow up with the necessary changes in the switchboard plug once we're happy with this.

I'm hoping this will help with resolving a lot of sleep/resume problems people have been experiencing, especially with laptops with external displays and with cases of devices suspending when they shouldn't. This does true late locking now and stays in your user session, not switching to the greeter until there's activity, rather than switching instantly to the lightdm session running as a different user.

Plus there's now a nice fade out animation (300ms on manual lock and 3 seconds on session timeout, which can be interrupted). This also allows music to continue playing in the background when the screensaver comes on and gives us the potential to display stuff on the "lock screen" in future if we wish.
